src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -104,6 +105,14 @@ var ReactDOMInput = {
         );
         didWarnCheckedLink = true;
       }
+      if (props.value !== undefined && props.defaultValue !== undefined && !didWarnValDefaultVal) {
+        warning(
+          false,
+          'Both `value` and `defaultValue` props on `input` should not be specified. ' +
```

Comment:
The first sentence in this error message makes it sound like neither of the props should be specified.  Alternative wording:

 > Input elements must be either controlled or uncontrolled (specify either the `value` prop, or the `defaultValue` prop, but not both).  Decide between using a controlled or uncontrolled input and remove one of these props.  More info: https://fb.me/react-controlled-components